spec/controllers/recordings_controller_spec.rb

```diff
@@ -53,6 +53,102 @@
     end
   end
 
+  describe '#update' do
+    # Metaprogramming -> mocking underlying method implemntation on the run.
+
+    let(:recording_update_params) do
+      {
+        name: 'Older'
+      }
+    end
+
+    let(:invalid_recording_update_params) do
+      {
+        name: ''
+      }
+    end
+
+    let(:pre_update_recordings_list) do
+      { returncode: true,
+        recordings: [{ recordID: 'd1784817183960eea1c2e5738eb8c62681f42e23-1654674116107',
+                       meetingID: 'jpykqjw7jvbe3s7dgxdsdl5dkmajusro62qfmzff',
+                       name: 'Newer',
+                       published: true,
+                       protected: 'false',
+                       state: 'published',
+                       startTime: 'Wed, 08 Jun 2022 08:41:56 +0100'.to_datetime,
+                       endTime: 'Wed, 08 Jun 2022 08:42:38 +0100'.to_datetime,
+                       participants: '1',
+                       metadata: { 'bbb-origin': 'greenlight', 'bbb-origin-version': '3', isBreakout: 'false',
+                                   'bn-presenter-name': 'Presenter' },
+                       playback: { format: { type: 'presentation',
+                                             url: 'https://blindsidenetworks.com/',
+                                             length: 1 } } }],
+        messageKey: '',
+        message: '' }
+    end
+
+    before do
+      room = create(:room, user:)
+      room.meeting_id = 'jpykqjw7jvbe3s7dgxdsdl5dkmajusro62qfmzff'
+      room.save
+
+      fake_bbb_api = Class.new(BigBlueButtonApi) do
+        def update_recordings(*_args)
+          BigBlueButtonApi::RECORDINGS_LIST[:recordings] = [
+            { recordID: 'd1784817183960eea1c2e5738eb8c62681f42e23-1654674116107',
+              meetingID: 'jpykqjw7jvbe3s7dgxdsdl5dkmajusro62qfmzff',
+              name: 'Newer',
+              published: true,
+              protected: 'false',
+              state: 'published',
+              startTime: 'Wed, 08 Jun 2022 08:41:56 +0100'.to_datetime,
+              endTime: 'Wed, 08 Jun 2022 08:42:38 +0100'.to_datetime,
+              participants: '1',
+              metadata: { 'bbb-origin': 'greenlight', 'bbb-origin-version': '3', isBreakout: 'false',
+                          'bn-presenter-name': 'Presenter', 'recording-name': 'Older' },
+              playback: { format: { type: 'presentation',
+                                    url: 'https://blindsidenetworks.com/',
+                                    length: 1 } } }
+          ]
```

Comment:
This isn't needed - we can just mock the response from the BigBlueButtonApi (See `recordings_sync_service`)

Reply:
The problem that I had was the fallowing:
  The update recording logic is a bit different where the source of truth isn't Greenlight but rather the **BBB/SL** server until a` r-sync` request gets issued to synchronize **GL**.
  That means upon testing and when issuing an update, **GL** will send the update API call to **BBB/SL** but nothing will reflect on our side except that the operation was successful meaning that **GL** copy of recordings will not include the update leaving me unable to make the functional test.
  I found that I need a way to mock both of the **API** calls that r-sync and update logics uses in a way that the mocks share a state that the update will change so that the second `r-sync` reflect the changes in **GL** ending up updating the table so that I can complete the test otherwise I'll be mocking the end result before making any updates.
  This was the cleanest way to mock the underlying implementation of `BBB_API `not only its returned value so that I can trigger a change on the `get_recordings `result via `update_recordings`. 
  -> I feel that there may be something a lot simpler than this but couldn't simply find it.